crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -929,6 +929,119 @@ impl Measured2d for Annulus {
+    fn default() -> Self {
+        Self {
+            half_diagonals: Vec2::splat(0.5),
+        }
+    }
+}
+
+impl Rhombus {
+    /// Create a new `Rhombus` from a vertical and horizontal diagonal sizes.
+    #[inline(always)]
+    pub fn new(horizontal_diagonal: f32, vertical_diagonal: f32) -> Self {
+        Self {
+            half_diagonals: Vec2::new(horizontal_diagonal / 2.0, vertical_diagonal / 2.0),
+        }
+    }
+
+    /// Get the length of each side of the rhombus
+    #[inline(always)]
+    pub fn side(&self) -> f32 {
+        (self.half_diagonals.x.powi(2) + self.half_diagonals.y.powi(2)).sqrt()
+    }
+
+    /// Get the radius of the circumcircle on which all vertices
+    /// of the rhombus lie
+    #[inline(always)]
+    pub fn circumradius(&self) -> f32 {
+        self.half_diagonals.x.max(self.half_diagonals.y)
+    }
+
+    /// Get the radius of the largest circle that can
+    /// be drawn within the rhombus
+    #[inline(always)]
+    #[doc(alias = "apothem")]
+    pub fn inradius(&self) -> f32 {
+        let horizontal_diagonal = self.half_diagonals.x * 2.0;
+        let vertical_diagonal = self.half_diagonals.y * 2.0;
+        (horizontal_diagonal * vertical_diagonal)
+            / (2.0 * (horizontal_diagonal.powi(2) + vertical_diagonal.powi(2)).sqrt())
+    }
+
+    /// Finds the point on the rhombus that is closest to the given `point`.
+    ///
+    /// If the point is outside the rhombus, the returned point will be on the perimeter of the rhombus.
+    /// Otherwise, it will be inside the rhombus and returned as is.
+    #[inline(always)]
+    pub fn closest_point(&self, point: Vec2) -> Vec2 {
+        let diagonals_mult = self.half_diagonals.x * self.half_diagonals.y;
+
+        // Point is inside the rhombus
+        if point.x.abs() * self.half_diagonals.y + point.y.abs() * self.half_diagonals.x
+            <= diagonals_mult
+        {
+            point
+        } else {
+            // Equation of the line that contains the side in the points quadrant
+            let [m, b] = [
+                (-point.x * point.y).signum() * self.half_diagonals.y / self.half_diagonals.x,
+                point.y.signum() * self.half_diagonals.y,
+            ];
+
+            let inv_m = -1.0 / m;
+
+            // Normal of the line mx+b that passes through the point
+            let side_normal = [inv_m, point.y - inv_m * point.x];
+
+            // Intersection of the side normal and the side line rounded to 4 decimal places
+            let xintercept =
+                (((b - side_normal[1]) / (side_normal[0] - m)) * 10000.0).round() / 10000.0;
+            let yintercept = ((m * xintercept + b) * 10000.0).round() / 10000.0;
+
+            if xintercept.abs() * self.half_diagonals.y + yintercept.abs() * self.half_diagonals.x
+                <= diagonals_mult
+            {
+                // The interception betwen the side and the normal is in the perimeter of the rhombus
```

Comment:
```suggestion
                // The interception between the side and the normal is in the perimeter of the rhombus
```

Reply:
Thank you, I think we solved this in the recent commit